content/stdout-vs-stderr.md

```diff
@@ -0,0 +1,2085 @@
+
+That is most likely why I was drawn into building terminal user interface applications with Rust in the first place. When I built my first ever Rust/TUI project, [kmon](https://github.com/orhun/kmon), I was surprised by how the limits of a simple thing such as a terminal can be pushed to build applications which gets you addicted to using terminals even more. Couple of years later, I'm one of the maintainers of [Ratatui](https://ratatui.rs/) which is a Rust library for cooking up TUIs and I'm blessed to be one of the core team members which revived the unmaintained [tui-rs](https://github.com/fdehau/tui-rs) library as Ratatui last year.
+
+When you take all of this into account, as a daily terminal user and a command-line developer, I'm tackling new terminal related issues every day. Sometimes I come across really interesting questions and problems. As you might expect, this blog post is the fruit of one of those questions.
+
+> <glitched>Why stdout is faster than stderr?</glitched>
+
+Okay, now let's take a step back and try to understand the question first. We need to grasp some concepts about UNIX before everything.
+
+<details>
+<summary><b>Table of Contents</b></summary>
+
+- [I/O Streams](#i-o-streams)
+- [TUI Applications and I/O](#tui-applications-and-i-o)
+- [Measuring FPS](#measuring-fps)
+- [Profiling](#profiling)
+- [Testing the buffered theory](#testing-the-buffered-theory)
+- [Experimenting with buffered writes](#experimenting-with-buffered-writes)
+- [Experimenting with raw writes](#experimenting-with-raw-writes)
+- [Making stdout faster](#making-stdout-faster)
+- [Findings](#findings)
+- [Other Languages](#other-languages)
+- [Conclusion](#conclusion)
+
+</details>
+
+---
+
+## <g>I/O Streams</g>
+
+[UNIX](https://en.wikipedia.org/wiki/Unix) operating system brought many groundbreaking advances into the world of computers and undoubtedly one of them was the standard streams. According to UNIX, every process has three streams opened for it when it starts up:
+
+0\. Standard Input (`stdin`): for reading input.  
+1\. Standard Output (`stdout`): for writing conventional output.  
+2\. Standard Error (`stderr`): for printing diagnostic or error messages.
+
+Here is an oversimplified example to demonstrate these streams:
+
+```bash
+# read the value of foo from stdin
+$ read -r foo
+test
+
+# print the value of var to stdout
+$ echo "value of foo is '$var'"
```

Comment:
```suggestion
# print the value of foo to stdout
$ echo "value of foo is '$foo'"
```

Reply:
It's better to keep the same var name here

Reply:
good catch, thanks!